spec/datadog/core/telemetry/logging_spec.rb

```diff
@@ -82,6 +82,126 @@ def log!(_event)
+          expect(event.payload).to include(
+            logs: [{message: 'Datadog::Core::Native::RuntimeError: (This is a safe profiler error)', level: 'ERROR', count: 1,
+                    stack_trace: a_string_including('REDACTED')}]
+          )
+        end
+        begin
+          raise Datadog::Core::Native::RuntimeError.new('This is a safe profiler error', 'This is a safe profiler error')
+        rescue => e
+          component.report(e, level: :error)
+        end
+      end
+      context 'with description' do
+        it 'includes both description and telemetry message' do
+          expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+            expect(event.payload).to include(
+              logs: [{message: 'Datadog::Core::Native::RuntimeError: Profiler failed to start (Failed to initialize native extension)', level: 'ERROR', count: 1,
+                      stack_trace: a_string_including('REDACTED')}]
+            )
+          end
+          begin
+            raise Datadog::Core::Native::RuntimeError.new('Failed to initialize native extension', 'Failed to initialize native extension')
+          rescue => e
+            component.report(e, level: :error, description: 'Profiler failed to start')
+          end
+        end
+      end
+      context 'without telemetry message' do
+        it 'omits the dynamic exception message from telemetry' do
+          expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+            expect(event.payload).to include(
+              logs: [{message: 'Datadog::Core::Native::RuntimeError', level: 'ERROR', count: 1,
+                      stack_trace: a_string_including('REDACTED')}]
+            )
+            expect(event.payload[:logs].map { |log| log[:message] }).not_to include(/Failed to initialize.*0x[0-9a-f]+/)
+          end
+          begin
+            raise Datadog::Core::Native::RuntimeError, 'Failed to initialize string storage: Error at address 0xdeadbeef'
+          rescue => e
+            component.report(e, level: :error)
+          end
+        end
+      end
+      context 'with telemetry message and dynamic content' do
+        it 'includes only the telemetry-safe message' do
+          expect(component).to receive(:log!).with(instance_of(Datadog::Core::Telemetry::Event::Log)) do |event|
+            expect(event.payload).to include(
+              logs: [{message: 'Datadog::Core::Native::RuntimeError: (Static message)', level: 'ERROR', count: 1,
+                      stack_trace: a_string_including('REDACTED')}]
+            )
+            expect(event.payload[:logs].map { |log| log[:message] }).not_to include('Dynamic info 0xabc123')
+          end
+          begin
+            raise Datadog::Core::Native::RuntimeError.new('Dynamic info 0xabc123', 'Static message')
+          rescue => e
```

Comment:
This seems to be testing the same as `it 'includes the telemetry-safe message in telemetry'` or am I missing something? 🤔 

I guess `description` is missing in `report`... maybe?